data_structures/linked_list_find_middle.cpp

```diff
@@ -0,0 +1,149 @@
+/**
+ * @file
+ * @brief A linked list implementation using nodes to find the middle element
+ * @warning This program does not use a sort function to order the linked list, order is based on original data entry
+ */
+
+#include <iostream>
+#include <cassert>
+#include <array>
+
+/**
+* @brief Structure for Node
+*/
+struct Node
+{
+    Node* next; ///< forward reference
+    int val;    ///< variable that holds the value of the node    
+};
+
+/**
+* @brief Create linked list and add nodes
+* @param Node** head passes in a Node* by reference
+* @param valueIn is the value of the node to be added
+* @returns void
+*/
+void addNode(struct Node** head, int valueIn)
+{
+    
+    struct Node* ptr = new Node; /*< Create new Node to store in linked list */
+    ptr->val = valueIn;
+    ptr->next = nullptr;
+    //Points new node's next to the head and makes the new node the head of the linked list
+    if(*head != nullptr)
+    {
+        ptr->next = *head;
+        *head = ptr;
+    }
+    //If head is nullptr, this is the new head of the list
+    else
+    {
+        *head = ptr;
+    }
+}
+
+/**
+* @brief Tortoise and Hare algorithm utilizing a fast and slow pointer
+* @returns Middle element of the linked list, if even, returns element one position to the right of the first middle value
+* @param struct Node** head passes in a Node* by reference
+*/
+int midPoint(struct Node** head)
+{
+    //create two nodes and set them at the head of the linked list
+    struct Node* slowptr = *head;
+    struct Node* fastptr = *head;
+    //ensure loop runs until fastptr reaches the end of the linked list
+    //when fastptr reaches the end of the list than slowptr will be at
+    //the middle element of the linked list
+    while(fastptr && fastptr->next && fastptr->next->next)
+    {
+        //move by 1 for slowptr
+        slowptr = slowptr->next;
+        //move by 2 for fastptr
+        fastptr = fastptr->next->next;
+    }
+    //store value of middle node and return the integer value
+    int ans = slowptr->val;
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. Check other parts of the code ([reference](https://en.cppreference.com/w/cpp/types/integer)). **Let us know if you need any help here or in our [Discord server](https://discord.gg/c7MnfGFGa6).** 🙂

Reply:
I will use the uint32_t in my next commit and specify to users, let me know if it checks out.